server/etcdmain/grpc_proxy.go

```diff
@@ -352,6 +362,24 @@ func newClientCfg(lg *zap.Logger, eps []string) (*clientv3.Config, error) {
 		cfg.TLS = clientTLS
 		lg.Info("gRPC proxy client TLS", zap.String("tls-info", fmt.Sprintf("%+v", tls)))
 	}
+
+	if grpcUser != "" {
+		var err error
+		if grpcPassword == "" {
+			splitted := strings.SplitN(grpcUser, ":", 2)
+			if len(splitted) < 2 {
+				cfg.Username = grpcUser
+				cfg.Password, err = speakeasy.Ask("Password: ")
```

Comment:
Just a quick comment (I'm still reading the change though): getting password in a prompt wouldn't be suitable for server side components. If it's ok for you I suggest to drop this feature. How do you think?

Reply:
well, I will rid it.  I copy it from etcdctl, haha >_<

Reply:
done